src/templates/bigquery.py

```diff
@@ -38,74 +18,125 @@
     cast,
 )
 
+import neo4j
+import neo4j.exceptions
+import pyarrow as pa
+from dataproc_templates import BaseTemplate
+from google.protobuf import descriptor_pb2
+from graphdatascience.query_runner import gds_arrow_client
+from graphdatascience.query_runner.gds_arrow_client import GdsArrowClient
+from pandas import DataFrame
+from pyarrow import Table, RecordBatch
+from pyspark.sql import SparkSession
+
+import pattern_parser
+from model import Edge, Node, arrow_to_nodes, arrow_to_edges
+from . import constants as c, util
+from .bq_client import BigQuerySource, BigQuerySink, BQStream
+from .mapper import node_mapper, edge_mapper, MappingFn
+from .model import Graph
+
 __all__ = [
     "BigQueryToNeo4jGDSTemplate",
     "Neo4jGDSToBigQueryTemplate",
 ]
 
 Arrow = Union[pa.Table, pa.RecordBatch]
+ArrowData = Union[pa.Table,  Iterable[pa.RecordBatch]]
+DataStream = Generator[Arrow, None, None]
+
 
+class ProgressTracker:
+    def __init__(self):
+        self.progress = 0
 
-def load_model_from_gcs(uri: str) -> Optional[na.model.Graph]:
+    def callback(self, progress: int) -> None:
+        self.progress = progress
+
+
+def load_model_from_gcs(uri: str) -> Optional[Graph]:
     """
     Attempt to load a Graph model from a GCS uri. Returns None on failure.
     """
     try:
         import fsspec
 
         with fsspec.open(uri, "rt") as f:
-            return na.model.Graph.from_json(f.read())
+            return Graph.from_json(f.read())
     except Exception:
         return None
 
 
 def flatten(
-    lists: List[List[Any]], fn: Optional[Callable[[Any], Any]] = None
+        lists: List[List[Any]], fn: Optional[Callable[[Any], Any]] = None
 ) -> List[Any]:
     """
     Flatten a list of lists, applying an optional function (fn) to the initial
     list of lists.
     """
     if not fn:
-
         def fn(x: Any) -> Any:
             return x
 
     return [x for y in map(fn, lists) for x in y]
 
 
+def map_tables(model: Graph, entity_type: str) -> Callable[[DataStream], List[ArrowData]]:
+    def map_entity(table: Arrow, mapper: MappingFn):
```

Comment:
This function seems like internal, so maybe worth prefixing it with underscore as well?